Firestore/core/src/firebase/firestore/util/async_queue.cc

```diff
@@ -104,6 +136,9 @@ void AsyncQueue::VerifySequentialOrder() const {
 
 void AsyncQueue::EnqueueBlocking(const Operation& operation) {
   VerifySequentialOrder();
+  if (is_shutting_down_) {
```

Comment:
Note: `EnqueueBlocking` is a test-only function; I'm not sure it needs to be aware of `is_shutting_down_` (which also saves from the trouble with locking here).

Reply:
Done.